[dunliang0513] iP

dunliang0513 opened the pull request.

Commit:
Add Gradle support

dunliang0513 committed:
Greet, Echo, Exit

dunliang0513 committed:
Add, List

dunliang0513 committed:
Mark as Done

dunliang0513 committed:
Todos, Events, Deadlines

dunliang0513 committed:
Handle Errors

dunliang0513 committed:
Level 6 Delete. Add support for deleting tasks from the list.

dunliang0513 committed:
Save the tasks in the hard disk automatically whenever the task list changes

dunliang0513 committed:
Teach Duke how to understand dates and time

dunliang0513 committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Save the tasks in the hard disk automatically whenever the task list changes

dunliang0513 committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Teach Duke how to understand dates and time

dunliang0513 committed:
Makes the code more OOP.

dunliang0513 committed:
Divide classes into packages and organise them into suitable java packages.

dunliang0513 committed:
Creates a JAR File.

dunliang0513 committed:
Adds JavaDoc Comments

dunliang0513 committed:
Follow the coding standard.

dunliang0513 committed:
Adds Find method.

Tohtoroo commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,49 @@
+
+import Exceptions.CommandNotFoundException;
+
+import java.io.IOException;
```
Your import orders are slightly off due to not being in alphabetical order, and the java.io imports should be above the duke imports!

Tohtoroo reviewed (commented):
Overall, the code was quite easy to read, just that maybe the import orders are slightly off. Possibly could switch up your Ui class to hold more functions for the interface, as it seems like most of your strings are printed from your other classes. Remember to start adding JavaDoc comments!

antonlee59 commented on src/main/java/Duke/Tasks/Event.java:
```diff
@@ -0,0 +1,61 @@
+import java.time.format.DateTimeParseException;
+
+public class Event extends Task {
+    protected String startingTime;
+    protected String endingTime;
+    protected LocalDate localStartingDate;
+    protected LocalDate localEndingDate;
```
Nice naming of fields. Makes it very clear what each field is referring to.

antonlee59 commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,145 @@
+    }
+
+    public void mark(int input) {
+        Task currTask = this.taskList.get(input - 1);
+        currTask.markDone();
+        System.out.println("\t--------------------------");
+        System.out.println("\tNice! I've marked this task as done:");
+        System.out.println("\t\t" + currTask.toString());
+        System.out.println("\t--------------------------");
```
Perhaps, all these println can be abstracted out into the Ui.java class to separate the concerns of each class better.

antonlee59 reviewed (commented):
Overall, LGTM. Maybe you can try to futher abstract out your methods especially when it comes to the Ui display as most of the printing still isn't being done in the Ui class. Good job and all the best in your upcoming tasks!

dunliang0513 committed:
Revert "Adds Find method."

This reverts commit 1ff03d22b738c5c38fe29e5eef2385e0acd6613a.

dunliang0513 committed:
Fix bug

dunliang0513 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'
Merge the gradle support to the main branch so that it can support the A-Gradle, A-JUnit features.

dunliang0513 committed:
Restore all the code in src after accidentally remove them

dunliang0513 committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-A-JavaDoc

dunliang0513 committed:
Add gradle support and arrange all the code into a main folder

dunliang0513 committed:
Fix the code issue in the main branch

dunliang0513 committed:
Setup the GUI of the chatbot but not completely done yet

dunliang0513 committed:
Implement GUI successfully

dunliang0513 committed:
Merge branch 'branch-A-CodingStandard'

* branch-A-CodingStandard:
  Revert "Adds Find method."

# Conflicts:
#	src/main/java/Duke/Parser.java
#	src/main/java/Duke/TaskList.java

dunliang0513 committed:
Merge branch 'branch-A-JavaDoc'

* branch-A-JavaDoc:
  Add gradle support and arrange all the code into a main folder
  Fix bug

# Conflicts:
#	.gitignore
#	META-INF/MANIFEST.MF
#	build.gradle
#	gradlew.bat

dunliang0513 committed:
Assertions Methods: Document important assumptions in the code

This commit introduces a new function to the codebase called "Assertions".

These assertions document key assumptions that must hold in order for the function to work correctly. By using assertions, we can more easily catch bugs and errors that might occur due to unexpected conditions or inputs.

To use assertions, I added several new lines of code to the relevant functions in the codebase. These assertions document key assumptions that must hold in order for the function to work correctly. If the assertion fails, the program will immediately halt and raise an error, allowing us to quickly identify the source of the problem.

This commit should improve the reliability and robustness of the codebase by making important assumptions more explicit and easier to check.

dunliang0513 committed:
Improve my code quality by following closely to the 'Code Quality' topics that I have learned.

dunliang0513 committed:
Implement "Reminders" feature

dunliang0513 committed:
Merge branch 'master' into branch-A-CodeQuality

dunliang0513 committed:
Merge pull request #2 from dunliang0513/branch-A-CodeQuality

Improve my code quality by following closely to the 'Code Quality' to…

dunliang0513 committed:
Merge branch 'master' into branch-BCD-Extension

dunliang0513 committed:
Merge pull request #3 from dunliang0513/branch-BCD-Extension

Implement "Reminders" feature

dunliang0513 committed:
Merge branch 'master' into branch-A-Assertions

dunliang0513 committed:
Merge pull request #4 from dunliang0513/branch-A-Assertions

Assertions Methods: Document important assumptions in the code

dunliang0513 committed:
Finalise the features and add the Ui.png into the docs folder

dunliang0513 committed:
Update README

dunliang0513 committed:
Update the link of the .jar file

dunliang0513 committed:
Update README on the layout and the image website link

dunliang0513 committed:
Debug and clean the code

dunliang0513 committed:
Update the README according to the latest version of the jar file

dunliang0513 committed:
Update README.md

dunliang0513 committed:
Update README.md
